File: Python/AES256.py
from hashlib import md5
from base64 import b64decode, b64encode
from Crypto.Cipher import AES
from Crypto.Random import get_random_bytes
from Crypto.Util.Padding import pad, unpad

class AESCipher:

    def init (self, key):
        self.key = md5(key.encode('utf8')).hexdigest().encode('utf-8')

    def encrypt(self, data, iv):
        padded_data = pad (data.encode(), AES.block_size)
        aes_cipher = AES.new(self.key, AES.MODE_CBC, iv)
        encrypted_data = aes_cipher.encrypt(padded_data)
        return b64encode(encrypted_data)

    def decrypt(self, data, iv):
        encrypted_data = b64decode(data)
        aes_cipher = AES.new(self.key, AES.MODE_CBC, iv)
        return unpad(aes_cipher.decrypt(encrypted_data), AES.block_size)

File: Typescript/AES256.ts
import CryptoJS from 'crypto-js';


class AESCipher {
    readonly blockSize: number = 16;
    key: string = '';

    constructor(key: string) {
        this.key = CryptoJS.MD5(CryptoJS.enc.Utf8.parse(key)).toString();
    }

    generateRandomHex = (length: number): string => [...Array(length)].map(() => Math.floor(Math.random() * 16).toString(16)).join('');
    createIv = () => this.generateRandomHex(this.blockSize);

    encrypt = (data: string, iv: string | undefined) => {
        const encodedKey = CryptoJS.MD5(CryptoJS.enc.Utf8.parse(this.key)).toString();

        const usedIV = iv === undefined ? this.createIv() : iv
        const encodedIv = CryptoJS.enc.Utf8.parse(usedIV);

        const encrypted = CryptoJS.AES.encrypt(data, encodedKey, { iv: encodedIv, mode: CryptoJS.mode.CBC});

        return {
            data: encrypted.toString(),
            iv: encodedIv
        };
    }

    decrypt = (data: string, iv: string): string => {
        const encodedIv = CryptoJS.enc.Utf8.parse(iv);
        const decrypted = CryptoJS.AES.encrypt(data, this.key, { iv: encodedIv, mode: CryptoJS.mode.CBC});

        return decrypted.toString();
    }

}

export default AESCipher;